indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisorStateTest.java

```diff
@@ -2563,6 +2565,77 @@ public LagStats computeLagStats()
     EasyMock.verify(executorService, spec);
   }
 
+  @Test
+  public void test_calculateWorkerThreads_shouldHonourWorkerConfig()
+  {
+    final int totalThreads = 5;
+    SeekableStreamSupervisorTuningConfig tuningConfig = getTuningConfigWithWorkerThreads(totalThreads);
+    SeekableStreamSupervisorIOConfig ioConfig = getSupervisorIOConfig(1, null);
+    Assert.assertEquals(totalThreads, SeekableStreamSupervisor.calculateWorkerThreads(tuningConfig, ioConfig));
+  }
+
+  @Test
+  public void test_calculateWorkerThreads_shouldUseDefaultWorkerThreads()
+  {
+    SeekableStreamSupervisorTuningConfig tuningConfig = getTuningConfigWithWorkerThreads();
+    SeekableStreamSupervisorIOConfig ioConfig = getSupervisorIOConfig(1, null);
+    Assert.assertEquals(
+        DEFAULT_WORKER_THREADS,
+        SeekableStreamSupervisor.calculateWorkerThreads(tuningConfig, ioConfig)
+    );
+  }
+
+  @Test
+  public void test_calculateWorkerThreads_shouldUseMinimumWorkerThreadstWithTasks()
+  {
+    SeekableStreamSupervisorTuningConfig tuningConfig = getTuningConfigWithWorkerThreads();
+    SeekableStreamSupervisorIOConfig ioConfig = getSupervisorIOConfig(7, null);
+    Assert.assertEquals(
+        DEFAULT_WORKER_THREADS,
+        SeekableStreamSupervisor.calculateWorkerThreads(tuningConfig, ioConfig)
+    );
+  }
+
+  @Test
+  public void test_calculateWorkerThreads_shouldUseFactorOfTaskCount()
+  {
+    SeekableStreamSupervisorTuningConfig tuningConfig = getTuningConfigWithWorkerThreads();
+    SeekableStreamSupervisorIOConfig ioConfig = getSupervisorIOConfig(18, null);
+    Assert.assertEquals(
+        4,
+        SeekableStreamSupervisor.calculateWorkerThreads(tuningConfig, ioConfig)
+    );
+  }
+
+  @Test
+  public void test_calculateWorkerThreads_shouldUseAutoScalerConfig()
+  {
+    SeekableStreamSupervisorTuningConfig tuningConfig = getTuningConfigWithWorkerThreads();
+    AutoScalerConfig autoScalerConfig = new LagBasedAutoScalerConfig(
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        21,
+        null,
+        5,
+        null,
+        null,
+        true,
+        null,
+        null
+    );
+    SeekableStreamSupervisorIOConfig ioConfig = getSupervisorIOConfig(1, autoScalerConfig);
+    Assert.assertEquals(
+        5,
```

Comment:
This would then become:

```suggestion
        taskCountMax / defaultTasksPerThread,
```